aws-rds-optiongroup/src/test/java/software/amazon/rds/optiongroup/AbstractTestBase.java

```diff
@@ -181,4 +173,8 @@ protected ProgressEvent<ResourceModel, CallbackContext> invokeHandleRequest(
     ) {
         return getHandler().handleRequest(getProxy(), request, context, getProxyClient(), logger);
     }
+
+    protected void setupResourceModelsForTests() {
```

Comment:
Question: why do we need this function? My assumption is that by default option group name is null in the model. Is it incorrect?

Reply:
My observation is that the default value is null. But it gets overridden by the random OptionGroupName generated by each individual test. Then subsequent calls using RESOURCE_MODEL fail due to there being an OptionGroupName present.

Please let me know if you have a better solution to get around this issue or if you have any follow up questions.

Reply:
I have a slight concern that OptionGroup unit tests have side effects on Resource models shared by tests. This means that we are testing resource models that might differ from what we expect.

One option to fix this is to use a getter function that will create a new object every time it is invoked.